pallets/loans-ref/src/lib.rs

```diff
@@ -0,0 +1,672 @@
+#![cfg_attr(not(feature = "std"), no_std)]
+
+use cfg_primitives::Moment;
+use cfg_traits::{
+	ops::{EnsureDiv, EnsureFixedPointNumber, EnsureInto, EnsureMul, EnsureSub},
+	InterestAccrual, Permissions, PoolInspect, PoolReserve,
+};
+use cfg_types::{
+	adjustments::Adjustment,
+	permissions::{PermissionScope, PoolRole, Role},
+};
+use codec::{Decode, Encode, MaxEncodedLen};
+use frame_support::{
+	traits::{
+		tokens::{
+			self,
+			nonfungibles::{Inspect, Mutate, Transfer},
+		},
+		PalletError, UnixTime,
+	},
+	transactional, RuntimeDebug, StorageHasher,
+};
+use pallet::*;
+//use pallet_pool_system::pool_types::PoolLocator;
+use scale_info::TypeInfo;
+use sp_arithmetic::traits::checked_pow;
+use sp_runtime::{
+	traits::{BadOrigin, One},
+	ArithmeticError, FixedPointNumber, FixedPointOperand,
+};
+
+const SECONDS_PER_DAY: Moment = 3600 * 24;
+const SECONDS_PER_YEAR: Moment = SECONDS_PER_DAY * 365;
+
+/// Diferent write off status that an unhealthy loan can have
+#[derive(Encode, Decode, Clone, TypeInfo, MaxEncodedLen)]
+pub enum WriteOffStatus<Rate> {
+	/// Usual write off
+	Standard {
+		/// Index in the vec of write off groups.
+		write_off_index: u32,
+	},
+	/// Written off by an admin
+	Admin {
+		/// Percentage of outstanding debt we are going to write off on a loan
```

Comment:
Write-offs don't impact the debt, only the valuation
```suggestion
		/// Percentage of present value we are going to write off on a loan
```